scripts/add_new_region.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+OLD_REGION='us-east-1'
+
+LAYER_NAMES=("Datadog-Node10-x" "Datadog-Node12-x")
+NODE_VERSIONS_FOR_AWS_CLI=("nodejs10.x" "nodejs12.x")
+NEW_REGION=$1
+
+publish_layer() {
+    region=$1
+    layer_name=$2
+    aws_version_key=$3
+    layer_path=$4
+
+    version_nbr=$(aws lambda publish-layer-version --layer-name $layer_name \
+        --description "Datadog Lambda Layer for Node" \
+        --zip-file "fileb://$layer_path" \
+        --region $region \
+        --compatible-runtimes $aws_version_key \
+                        | jq -r '.Version')
+
+    aws lambda add-layer-version-permission --layer-name $layer_name \
+        --version-number $version_nbr \
+        --statement-id "release-$version_nbr" \
+        --action lambda:GetLayerVersion --principal "*" \
+        --region $region
+
+    echo "Published layer for region $region, version $aws_version_key, layer_name $layer_name, layer_version $version_nbr"
+}
+
+get_max_version() {
+    layer_name=$1
+    region=$2
+    last_layer_version=$(aws lambda list-layer-versions --layer-name $layer_name --region $region | jq -r ".LayerVersions | .[0] |  .Version")
+    echo $last_layer_version
+}
+
+if [ -z "$1" ]; then
+    echo "Region parameter not specified, exiting"
+    exit 1
+fi
+
+j=0
+for layer_name in "${LAYER_NAMES[@]}"; do
+    # get latest version
+    last_layer_version=$(get_max_version $layer_name $OLD_REGION)
+
+    # run for each version of layer
+    for i in $(seq 1 $last_layer_version); do 
+        layer_path=$layer_name"_"$i.zip
+        aws_version_key="${NODE_VERSIONS_FOR_AWS_CLI[$j]}"
+
+        # download layer versions 
+        URL=$(AWS_REGION=$OLD_REGION aws lambda get-layer-version --layer-name $layer_name --version-number $i --query Content.Location --output text)
```

Comment:
This is smart, although it might take a while to run.

Reply:
I think you only do this once when you're publishing to a new region so the delay should be okay considering this will ensure all the versions are correct.